types/directives.ts

```diff
@@ -0,0 +1,7 @@
+import { ObjectDirective, DirectiveBinding } from 'nuxt/dist/app/compat/capi';
+
+export type LoadingValue = boolean | undefined;
+
+export type LoadingBinding = DirectiveBinding<LoadingValue>;
+
+export type LoadingDirective = ObjectDirective<HTMLElement, LoadingValue>;
```

Comment:
Please tell me if what I put here is a good approach as this is my first serious attempt with TS

Reply:
looks good 👌